OSDOCS-12237-last-round (#89055)
openshift/openshift-docs: release_notes/ocp-4-18-release-notes.adoc

[id="ocp-release-notes-installation-and-update_{context}"]
=== Installation and update

[id="ocp-4-18-installation-cluster-api-provider-ibmcloud_{context}"]

> ==== New version of `cluster-api-provider-ibmcloud`
> 
> The installation program now uses a newer version of the `cluster-api-provider-ibmcloud` provider that includes Transit Gateway fixes. Because of the cost of Transit Gateways in {ibm-cloud-title}, you can now use the {product-title} to create a Transit Gateway when creating an {product-title} cluster. For more information, see (link:https://issues.redhat.com/browse/OCPBUGS-37588[*OCPBUGS-37588*]) and (link:https://issues.redhat.com/browse/OCPBUGS-41938[*OCPBUGS-41938*]).

Reviewer: There is an attribute for the correct name of this
```suggestion
==== New version of the {cap-ibm-short}

The installation program now uses a newer version of the {cap-ibm-short} that includes Transit Gateway fixes. Because of the cost of Transit Gateways in {ibm-cloud-title}, you can now use the {product-title} to create a Transit Gateway when creating an {product-title} cluster. For more information, see (link:https://issues.redhat.com/browse/OCPBUGS-37588[*OCPBUGS-37588*]) and (link:https://issues.redhat.com/browse/OCPBUGS-41938[*OCPBUGS-41938*]).
```

[id="ocp-release-notes-olm_{context}"]
=== Operator lifecycle

[id="ocp-release-notes-olm-classic-rename_{context}"]
==== Existing version of Operator Lifecycle Manager now known as {olmv0}

With the release of {olmv1-first} as a General Availability (GA) feature, starting in {product-title} {product-version}, the existing version of OLM that has been included since the launch of {product-title} 4 is now known as _{olmv0}_.

[NOTE]
====
{olmv0} remains enabled by default and fully supported throughout the {product-title} 4 lifecycle.
====

For more information on the GA release of {olmv1}, see the xref:../release_notes/ocp-4-18-release-notes.adoc#ocp-release-notes-extensions_release-notes[Extensions ({olmv1})] release note sections. For full documentation focused on {olmv1}, see the stand-alone xref:../extensions/index.adoc#olmv1-about[Extensions] guide.

For full documentation focused on {olmv0}, continue referring to the xref:../operators/index.adoc#operators-overview[Operators] guide.

> ////
> [id="ocp-release-notes-osdk_{context}"]
> === Operator development
> 
> 
> [id="ocp-release-notes-machine-management_{context}"]
> === Machine management
> ////

Reviewer: comment line in wrong spot ate the Machine management heading
```suggestion
////
[id="ocp-release-notes-osdk_{context}"]
=== Operator development
////

[id="ocp-release-notes-machine-management_{context}"]
=== Machine management
```

* Previously, a check for unexpected IP addresses on the provisioning interface during metal3 pod startup was triggered. This issue occurred because of the presence of an IP addresses supplied by DHCP from a previous version of the pod that existed on another node. With this release, a pod startup check now looks only for IP addresses that exist outside the provisioning network subnet, so that a metal3 pod starts immediately, even when if the node has moved to a different node. (link:https://issues.redhat.com/browse/OCPBUGS-38507[*OCPBUGS-38507*])

> * Previously, enabling a provisioning network by editing the cluster-wide `Provisioning` resource was only possible on installer-provisioned infrastructure clusters with platform type `baremetal`. On bare metal, single-node OpenShift (SNO) and user-provisioned infrastructure clusters resulted in a validation error. With this release, the excessive validation check has been removed, and enabling a provisioning network is now possible on bare-metal clusters with platform type `none`. As with installer-provisioned infrastructure clusters, users are responsible for making sure that all networking requirements are met for this operation. (link:https://issues.redhat.com/browse/OCPBUGS-43371[*OCPBUGS-43371*])

Reviewer: ```suggestion
* Previously, enabling a provisioning network by editing the cluster-wide `Provisioning` resource was only possible on installer-provisioned infrastructure clusters with platform type `baremetal`. On bare metal, {sno} and user-provisioned infrastructure clusters resulted in a validation error. With this release, the excessive validation check has been removed, and enabling a provisioning network is now possible on bare-metal clusters with platform type `none`. As with installer-provisioned infrastructure clusters, users are responsible for making sure that all networking requirements are met for this operation. (link:https://issues.redhat.com/browse/OCPBUGS-43371[*OCPBUGS-43371*])
```

* Previously, a check for unexpected IP addresses on the provisioning interface during metal3 pod startup was triggered. This issue occurred because of the presence of an IP addresses supplied by DHCP from a previous version of the pod that existed on another node. With this release, a pod startup check now looks only for IP addresses that exist outside the provisioning network subnet, so that a metal3 pod starts immediately, even when if the node has moved to a different node. (link:https://issues.redhat.com/browse/OCPBUGS-38507[*OCPBUGS-38507*])

> * Previously, enabling a provisioning network by editing the cluster-wide `Provisioning` resource was only possible on installer-provisioned infrastructure clusters with platform type `baremetal`. On bare metal, {sno} and user-provisioned infrastructure clusters resulted in a validation error. With this release, the excessive validation check has been removed, and enabling a provisioning network is now possible on bare-metal clusters with platform type `none`. As with installer-provisioned infrastructure clusters, users are responsible for making sure that all networking requirements are met for this operation. (link:https://issues.redhat.com/browse/OCPBUGS-43371[*OCPBUGS-43371*])

Reviewer: ```suggestion
* Previously, enabling a provisioning network by editing the cluster-wide `Provisioning` resource was only possible on installer-provisioned infrastructure clusters with platform type `baremetal`. On bare metal, {sno}, and user-provisioned infrastructure clusters, editing this resource resulted in a validation error. With this release, the excessive validation check has been removed, and enabling a provisioning network is now possible on bare-metal clusters with platform type `none`. As with installer-provisioned infrastructure clusters, users are responsible for making sure that all networking requirements are met for this operation. (link:https://issues.redhat.com/browse/OCPBUGS-43371[*OCPBUGS-43371*])
```

* Previously, enabling a provisioning network by editing the cluster-wide Provisioning resource was only possible on clusters with platform type `baremetal`, such as ones created by the IPI installer. On baremetal SNO and UPI clusters that would result in a validation error. The excessive validation has been removed, and enabling a provisioning network is now possible on baremetal clusters with platform type `none`. As with IPI, users are responsible for making sure that all networking requirements are met for this operation. (link:https://issues.redhat.com/browse/OCPBUGS-43371[*OCPBUGS-43371*])

> * Previously, the installation program populated the `network.devices`, `template` and `workspace` fields in the `spec.template.spec.providerSpec.value` section of the {vmw-full} control plane machine set custom resource (CR). These fields should be set in the {vmw-short} failure domain, and the installation program populating them caused unintended behaviors. Updating these fields did not trigger an update to the control plane machines, and these fields were cleared when the control plane machine set was deleted. With this release, the installation program is updated to no longer populate values that are included in the failure domain configuration. If these values are not defined in a failure domain configuration, for instance on a cluster that is updated to {product-title} {product-version} from an earlier version, the values defined by the installation program are used. (link:https://issues.redhat.com/browse/OCPBUGS-42660[*OCPBUGS-42660*])

Reviewer: ```suggestion
* Previously, the installation program populated the `network.devices`, `template`, and `workspace` fields in the `spec.template.spec.providerSpec.value` section of the {vmw-full} control plane machine set custom resource (CR). These fields should be set in the {vmw-short} failure domain, and the installation program populating them caused unintended behaviors. Updating these fields did not trigger an update to the control plane machines, and these fields were cleared when the control plane machine set was deleted. With this release, the installation program is updated to no longer populate values that are included in the failure domain configuration. If these values are not defined in a failure domain configuration, for instance on a cluster that is updated to {product-title} {product-version} from an earlier version, the values defined by the installation program are used. (link:https://issues.redhat.com/browse/OCPBUGS-42660[*OCPBUGS-42660*])
```

[id="ocp-release-note-installer-bug-fixes_{context}"]
==== Installer

> * Previously, when you attempted to set the policy for a {gcp-first} service account, the API reported a `400: Bad Request` validation error. When you create a service account, it might take up to 60 seconds for the account to become active, and this causes the validation error. If this error occurs, create a service account with a true exponential backoff that lasts at least 60 seconds. (link:https://issues.redhat.com/browse/OCPBUGS-48187[*OCPBUGS-48187*])

Reviewer: This has a workaround rather than a fix - seems like a known issue, not a bug fix

* Previously, on the *Observe -> Alerting -> Silences* tab, the `DateTime` component changed the ordering of an event and its value. Because of this issue, you could not edit the `until` parameter for a silent alert in either the *Developer* or the *Administrator* perspective. With this release, a fix means to the `DateTime` component means that you can now edit the `until` parameter for a silent alert. (link:https://issues.redhat.com/browse/OCPBUGS-46021[*OCPBUGS-46021*])

> * Previously, when using the *Developer* perspective with custom editors, clicking the `n` key caused the *Namespace* menu unexpectedly opened. The issue happened because the keyboard shortcut did not account for custom editors. With this release, the *Namespace* menu accounts for custom editors and does not open if you type the `n key. (link:https://issues.redhat.com/browse/OCPBUGS-38775[*OCPBUGS-38775*])

Reviewer: ```suggestion
* Previously, when using the *Developer* perspective with custom editors, clicking the `n` key caused the *Namespace* menu unexpectedly opened. The issue happened because the keyboard shortcut did not account for custom editors. With this release, the *Namespace* menu accounts for custom editors and does not open if you type the `n` key. (link:https://issues.redhat.com/browse/OCPBUGS-38775[*OCPBUGS-38775*])
```

* Previously, the `oc adm node-image create --pxe generated` command did not create only the Preboot Execution Environment (PXE) artifacts. Instead, the command created the PXE artifacts with other artifacts from a `node-joiner` pod and stored them all in the wrong subdirectory. Additionally, the PXE artifacts were incorrectly prefixed with `agent` instead of `node`. With this release, generated PXE artifacts are stored in the correct directory and receive the correct prefix. (link:https://issues.redhat.com/browse/OCPBUGS-46449[*OCPBUGS-46449*])

> * Previously, requests to the `deploymentconfig/scale` subresource would fail when there was an admission webhook matching the request. With this release, the issue is resolved and requests to the deploymentconfig/scale` subresource will succeed. (link:https://issues.redhat.com/browse/OCPBUGS-41136[*OCPBUGS-41136*]) 

Reviewer: ```suggestion
* Previously, requests to the `deploymentconfig/scale` subresource would fail when there was an admission webhook matching the request. With this release, the issue is resolved and requests to the `deploymentconfig/scale` subresource will succeed. (link:https://issues.redhat.com/browse/OCPBUGS-41136[*OCPBUGS-41136*]) 
```
